tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,437 @@
+
+  // Test surface_tangents
+
+  auto test_surface_tangents =
+      make_array<2>(Strahlkorper<Frame::Inertial>::ThreeVector(n_pts));
+  const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+
+  const auto theta = theta_phi.get(0);
+  const auto phi = theta_phi.get(1);
+  const auto cos_phi = cos(phi);
+  const auto sin_phi = sin(phi);
+  const auto cos_theta = cos(theta);
+  const auto sin_theta = sin(theta);
+
+  test_surface_tangents[0].get(0) =
+      cos_phi * cos_theta * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[0].get(1) =
+      cos_theta * sin_phi * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[0].get(2) =
+      amp * square(cos_theta) * sin_phi -
+      sin_theta * (radius + amp * sin_phi * sin_theta);
+  test_surface_tangents[1].get(0) =
+      -radius * sin_phi + amp * sin_theta * (2. * square(cos_phi) - 1);
+  test_surface_tangents[1].get(1) =
+      cos_phi * (radius + 2. * amp * sin_phi * sin_theta);
+  test_surface_tangents[1].get(2) = amp * cos_phi * cos_theta;
+
+  const auto& surface_tangents = strahlkorper.surface_tangents();
+  for (size_t i = 0; i < 2; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      for (size_t s = 0; s < n_pts; ++s) {
+        CHECK(gsl::at(surface_tangents, i).get(j)[s] ==
+              approx(gsl::at(test_surface_tangents, i).get(j)[s]));
+      }
+    }
+  }
+
+  // Test surface_cartesian_coordinates
+  Strahlkorper<Frame::Inertial>::ThreeVector test_cart_coords(n_pts);
+
+  {
+    const auto temp = radius + amp * sin_phi * sin_theta;
+    test_cart_coords.get(0) = cos_phi * sin_theta * temp + center[0];
+    test_cart_coords.get(1) = sin_phi * sin_theta * temp + center[1];
+    test_cart_coords.get(2) = cos_theta * temp + center[2];
+  }
+  const auto& cart_coords = strahlkorper.surface_cartesian_coords();
+  for (size_t j = 0; j < 3; ++j) {
+    for (size_t s = 0; s < n_pts; ++s) {
+      CHECK(test_cart_coords.get(j)[s] == approx(cart_coords.get(j)[s]));
+    }
+  }
+
+  // Test surface_normal_one_form
+  Strahlkorper<Frame::Inertial>::OneForm test_normal_one_form(n_pts);
+  {
+    const auto& r = strahlkorper.radius();
+    const auto temp = r + amp * sin_phi * sin_theta;
+    const auto one_over_r = 1.0 / r;
+    test_normal_one_form.get(0) = cos_phi * sin_theta * temp * one_over_r;
+    test_normal_one_form.get(1) =
+        (sin_phi * sin_theta * temp - amp) * one_over_r;
+    test_normal_one_form.get(2) = cos_theta * temp * one_over_r;
+  }
+  const auto& normal_one_form = strahlkorper.surface_normal_one_form();
+  for (size_t j = 0; j < 3; ++j) {
+    for (size_t s = 0; s < n_pts; ++s) {
+      CHECK(test_normal_one_form.get(j)[s] ==
+            approx(normal_one_form.get(j)[s]));
+    }
+  }
+
+  // Test surface_normal_magnitude.
+  Strahlkorper<Frame::Inertial>::InverseMetric invg(n_pts);
+  DataVector test_normal_mag(n_pts);
+
+  {
+    const auto& r = strahlkorper.radius();
+    invg.get(0, 0) = 1.0;
```

Comment:
Move `invg` initialization up to where it is defined (or at least above the declaration of `r`).